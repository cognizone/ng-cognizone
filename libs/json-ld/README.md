# json-ld

This library was generated with [Nx](https://nx.dev).

## Running unit tests

Run `nx test json-ld` to execute the unit tests.
