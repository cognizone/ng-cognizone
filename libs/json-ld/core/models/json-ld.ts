import { RDF } from '@cognizone/lod';
import { IRI_PATTERN, Primitive } from '@cognizone/model-utils';

export type JsonLd = JsonLdGraph | JsonLdNode[];

export interface JsonLdGraph {
  '@graph': JsonLdNode[];
  '@context': JsonLdContext;
}

export interface JsonLdNode {
  '@id': string;
  '@type': string[];
}

export interface JsonLdBlankNode {
  '@id': string;
}

export type JsonLdValue<T extends Primitive = Primitive> = JsonLdList | JsonLdValueLang | JsonLdValueLiteral<T> | JsonLdValueReference;

export interface JsonLdValueLiteral<T extends Primitive = string> {
  '@value': T;
  '@type'?: string;
}

export interface JsonLdValueLang {
  '@value': string;
  '@language': string;
}

export interface JsonLdValueReference {
  '@id': string;
}

export interface JsonLdList {
  '@list': JsonLdValue[];
}

export interface JsonLdContext {}

export interface JsonLdGraphCz<T = unknown> {
  data: JsonLdNode[];
  '@context': JsonLdContext;
  facets?: T;
}

export type JsonLdNodeFlat<T = JsonLdNode> = {
  [key in keyof T]: T[key] extends JsonLdNode
    ? JsonLdValueReference
    : T[key] extends JsonLdNode | undefined
    ? JsonLdValueReference | undefined
    : T[key] extends JsonLdNode[]
    ? JsonLdValueReference[]
    : T[key] extends JsonLdNode[] | undefined
    ? JsonLdValueReference[] | undefined
    : T[key];
};

export function isJsonLdValueLiteral<T extends Primitive>(o: unknown): o is JsonLdValueLiteral<T> {
  return typeof o === 'object' && o != null && '@value' in o;
}

export function isJsonLdValueLang(o: JsonLdValue): o is JsonLdValueLang {
  return '@language' in o;
}

export function isJsonLdList(o: JsonLdValue): o is JsonLdList {
  return '@list' in o;
}

export function isJsonLdValueReference(o: unknown): o is JsonLdValueReference {
  return typeof o === 'object' && o != null && '@id' in o;
}

export interface RdfListElement extends JsonLdNode {
  [RDF.first]: JsonLdValue[];
  [RDF.rest]?: JsonLdValue[];
}

export function isRdfListElement(value: unknown): value is RdfListElement {
  return typeof value === 'object' && value !== null && '@id' in value && RDF.first in value;
}

export function isBlankNodeUri(id: string): boolean {
  return !!id && (id.startsWith('_:') || !id.match(IRI_PATTERN));
}

export function isBlankNode(node: unknown): node is JsonLdValueReference {
  return isJsonLdValueReference(node) && isBlankNodeUri(node['@id']);
}
