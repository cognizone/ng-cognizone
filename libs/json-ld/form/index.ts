export * from './services';
export * from './directives';
export * from './json-ld-form.module';
