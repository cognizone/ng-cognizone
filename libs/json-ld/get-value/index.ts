export * from './json-ld-get-value.pipe';
