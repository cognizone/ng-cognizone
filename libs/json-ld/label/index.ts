export * from './json-ld-label.pipe';
