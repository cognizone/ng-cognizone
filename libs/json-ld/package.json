{
  "name": "@cognizone/json-ld",
  "version": "0.0.1",
  "peerDependencies": {
    "@angular/core": "^14.0.0",
    "@cognizone/lod": ">=3.0.0",
    "@cognizone/i18n": ">=3.0.0",
    "@cognizone/model-utils": ">=3.0.0",
    "@cognizone/ng-core": ">=3.0.0",
    "@ngxs/store": "^3.7.5",
    "immer": "^9.0.0",
    "json-ld": "^8.0.0",
    "lodash-es": "^4.17.21",
    "rxjs": "^7.0.0"
  },
  "dependencies": {
    "tslib": "^2.3.0"
  },
  "license": "Apache-2.0",
  "homepage": "https://github.com/cognizone/ng-cognizone",
  "contributors": [
    {
      "name": "William Chelman",
      "email": "william.chelman@cogni.zone"
    },
    {
      "name": "Farah El Agha",
      "email": "farah.agha111@gmail.com"
    }
  ],
  "repository": {
    "type": "git",
    "url": "https://github.com/cognizone/ng-cognizone.git",
    "directory": "libs/json-ld"
  },
  "private": false,
  "publishConfig": {
    "access": "public"
  }
}
