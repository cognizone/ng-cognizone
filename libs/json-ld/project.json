{
  "name": "json-ld",
  "$schema": "../../node_modules/nx/schemas/project-schema.json",
  "projectType": "library",
  "sourceRoot": "libs/json-ld/src",
  "prefix": "cz",
  "targets": {
    "build": {
      "executor": "@nrwl/angular:package",
      "outputs": ["{workspaceRoot}/dist/{projectRoot}"],
      "options": {
        "project": "libs/json-ld/ng-package.json"
      },
      "configurations": {
        "production": {
          "tsConfig": "libs/json-ld/tsconfig.lib.prod.json"
        },
        "development": {
          "tsConfig": "libs/json-ld/tsconfig.lib.json"
        }
      },
      "defaultConfiguration": "production"
    },
    "test": {
      "executor": "@nrwl/jest:jest",
      "outputs": ["{workspaceRoot}/coverage/{projectRoot}"],
      "options": {
        "jestConfig": "libs/json-ld/jest.config.ts",
        "passWithNoTests": true
      }
    },
    "lint": {
      "executor": "@nrwl/linter:eslint",
      "options": {
        "lintFilePatterns": ["libs/json-ld/**/*.ts", "libs/json-ld/**/*.html"]
      }
    }
  },
  "tags": []
}
