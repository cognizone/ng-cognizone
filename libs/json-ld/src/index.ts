export const name = 'json-ld';
