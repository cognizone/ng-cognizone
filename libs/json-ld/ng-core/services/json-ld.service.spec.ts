import { createServiceFactory, SpectatorService } from '@ngneat/spectator/jest';

import { JsonLdGraph } from '@cognizone/json-ld-core';
import { JsonLdService } from './json-ld.service';

describe('JsonLdService', () => {
  const createService = createServiceFactory({
    service: JsonLdService,
  });
  let spectator: SpectatorService<JsonLdService>;
  let service: JsonLdService;
  beforeEach(() => {
    spectator = createService();
    service = spectator.service;
  });

  it('should be created', () => {
    expect(service).toBeTruthy();
  });

  it('should expand json with facet', async () => {
    const result = await service.expand(json1);
    expect(result).toMatchSnapshot();
  });
});

const json1 = {
  data: [
    {
      '@id': 'https://example.org/app/cmdb',
      '@type': 'prov:SoftwareAgent',
      'dcterms:description': [
        {
          '@language': 'en',
          '@value': '??',
        },
      ],
      'dcterms:title': [
        {
          '@language': 'en',
          '@value': 'Configuration Management Database (CMDB)',
        },
      ],
      'adms:status': [
        {
          '@id': 'https://example.org/def/asset-status/inProduction',
        },
      ],
      'dcat:contactPoint': [
        {
          '@id': 'https://example.org/contact-point/document-management-unit',
        },
      ],
      'dcat:qualifiedRelation': [
        {
          '@id': 'https://example.org/relation/cmdb-gets-data-from-codict',
        },
      ],
      'prov:qualifiedAttribution': [
        {
          '@id': 'https://example.org/attribution/cmdb-application-owner',
        },
      ],
      'asset-relation:getsDataFrom': [
        {
          '@id': 'https://example.org/app/codict',
        },
      ],
    },
    {
      '@id': 'https://example.org/attribution/cmdb-data-data-owner',
      '@type': 'prov:Attribution',
      'dcat:hadRole': [
        {
          '@id': 'https://example.org/def/agent-role/dataOwner',
        },
      ],
      'prov:agent': [
        {
          '@id': 'https://example.org/org/itec-support',
        },
      ],
    },
    {
      '@id': 'https://example.org/contact-point/servicenow-support',
      '@type': 'http://www.w3.org/2006/vcard/ns#Kind',
      'http://www.w3.org/2006/vcard/ns#fn': [
        {
          '@language': 'en',
          '@value': 'ServiceNow Support',
        },
      ],
      'http://www.w3.org/2006/vcard/ns#hasEmail': [
        {
          '@id': 'mailto:servicenow-support@example.org',
        },
      ],
    },
    {
      '@id': 'https://example.org/dataset/cmdb-data',
      '@type': 'dcat:Dataset',
      'dcterms:accrualPeriodicity': [
        {
          '@id': 'https://example.org/def/frequency/continuous',
        },
      ],
      'dcterms:created': [
        {
          '@type': 'xsd:date',
          '@value': '2023-08-01',
        },
      ],
      'dcterms:description': [
        {
          '@language': 'en',
          '@value': 'CMDB stays for Configuration Management Database.',
        },
      ],
      'dcterms:issued': [
        {
          '@type': 'xsd:date',
          '@value': '2023-08-01',
        },
      ],
      'dcterms:modified': [
        {
          '@type': 'xsd:date',
          '@value': '2024-10-28',
        },
      ],
      'dcterms:title': [
        {
          '@language': 'en',
          '@value': 'Data from CMDB (Ticketing tool handling requests for IT equipment)',
        },
      ],
      'dcterms:type': [
        {
          '@id': 'https://example.org/def/dataset-type/master-data',
        },
      ],
      'adms:status': [
        {
          '@id': 'https://example.org/def/asset-status/inProduction',
        },
      ],
      'dcat:contactPoint': [
        {
          '@id': 'https://example.org/contact-point/servicenow-support',
        },
      ],
      'dcat:qualifiedRelation': [
        {
          '@id': 'https://example.org/relation/cmdb-data-stored-by-cmdb',
        },
      ],
      'dcat:theme': [
        {
          '@id': 'https://example.org/def/business-area/informationTechnology',
        },
      ],
      'prov:qualifiedAttribution': [
        {
          '@id': 'https://example.org/attribution/cmdb-data-data-owner',
        },
      ],
      'asset-relation:storedBy': [
        {
          '@id': 'https://example.org/app/cmdb',
        },
      ],
      'app:confidentialityLevel': [
        {
          '@id': 'https://example.org/def/confidentiality-level/snc',
        },
      ],
    },
    {
      '@id': 'https://example.org/def/agent-role/dataOwner',
      '@type': ['dcat:Role', 'skos:Concept'],
      'rdfs:label': [
        {
          '@language': 'en',
          '@value': 'data owner',
        },
      ],
      'skos:inScheme': [
        {
          '@id': 'https://example.org/def/agent-role',
        },
      ],
      'skos:preflabel': [
        {
          '@language': 'en',
          '@value': 'data owner',
        },
      ],
    },
    {
      '@id': 'https://example.org/def/asset-status/inProduction',
      '@type': 'skos:Concept',
      'skos:inScheme': [
        {
          '@id': 'https://example.org/def/asset-status',
        },
      ],
      'skos:prefLabel': [
        {
          '@language': 'en',
          '@value': 'In production',
        },
      ],
    },
    {
      '@id': 'https://example.org/def/business-area/informationTechnology',
      '@type': 'skos:Concept',
      'skos:inScheme': [
        {
          '@id': 'https://example.org/def/business-area',
        },
      ],
      'skos:prefLabel': [
        {
          '@language': 'en',
          '@value': 'Information Technology',
        },
      ],
    },
    {
      '@id': 'https://example.org/def/confidentiality-level/snc',
      '@type': 'skos:Concept',
      'skos:inScheme': [
        {
          '@id': 'https://example.org/def/confidentiality-level',
        },
      ],
      'skos:prefLabel': [
        {
          '@language': 'en',
          '@value': 'Sensitive non-confidential',
        },
      ],
    },
    {
      '@id': 'https://example.org/def/dataset-type/master-data',
      '@type': 'skos:Concept',
      'skos:inScheme': [
        {
          '@id': 'https://example.org/def/dataset-type',
        },
      ],
      'skos:prefLabel': [
        {
          '@language': 'en',
          '@value': 'Master Data',
        },
      ],
    },
    {
      '@id': 'https://example.org/def/frequency/continuous',
      '@type': 'skos:Concept',
      'skos:inScheme': [
        {
          '@id': 'https://example.org/def/frequency',
        },
      ],
      'skos:prefLabel': [
        {
          '@language': 'en',
          '@value': 'continuous',
        },
      ],
    },
    {
      '@id': 'https://example.org/org/itec-support',
      '@type': 'org:Organization',
      'skos:prefLabel': [
        {
          '@language': 'en',
          '@value': 'ITEC - SUPPORT',
        },
      ],
    },
    {
      '@id': 'https://example.org/relation/cmdb-data-stored-by-cmdb',
      '@type': 'dcat:Relationship',
      'dcterms:relation': [
        {
          '@id': 'https://example.org/app/cmdb',
        },
      ],
      'dcterms:type': [
        {
          '@id': 'asset-relation:storedBy',
        },
      ],
    },
  ],
  '@context': {
    '@vocab': 'https://example.org/def/dcat#',
    xsd: 'http://www.w3.org/2001/XMLSchema#',
    skos: 'http://www.w3.org/2004/02/skos/core#',
    rdfs: 'http://www.w3.org/2000/01/rdf-schema#',
    'asset-relation': 'https://example.org/def/asset-relation/',
    app: 'https://example.org/def/epdp#',
    dcterms: 'http://purl.org/dc/terms/',
    dcat: 'http://www.w3.org/ns/dcat#',
    prov: 'http://www.w3.org/ns/prov#',
    adms: 'http://www.w3.org/ns/adms#',
    org: 'http://www.w3.org/ns/org#',
    rdf: 'http://www.w3.org/1999/02/22-rdf-syntax-ns#',
    data: '@graph',
  },
  facets: {
    rootUri: 'https://example.org/dataset/cmdb-data',
    config: 'dataset',
    types: ['http://www.w3.org/ns/dcat#Dataset', 'http://www.w3.org/ns/dcat#Dataset'],
  },
} as unknown as JsonLdGraph;
