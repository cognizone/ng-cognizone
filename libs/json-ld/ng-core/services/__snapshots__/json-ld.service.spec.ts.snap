// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JsonLdService should expand json with facet 1`] = `
Object {
  "facets": Object {
    "config": "dataset",
    "rootUri": "https://example.org/dataset/cmdb-data",
    "types": Array [
      "http://www.w3.org/ns/dcat#Dataset",
      "http://www.w3.org/ns/dcat#Dataset",
    ],
  },
  "nodes": Object {
    "https://example.org/app/cmdb": Object {
      "@id": "https://example.org/app/cmdb",
      "@type": Array [
        "http://www.w3.org/ns/prov#SoftwareAgent",
      ],
      "http://purl.org/dc/terms/description": Array [
        Object {
          "@language": "en",
          "@value": "??",
        },
      ],
      "http://purl.org/dc/terms/title": Array [
        Object {
          "@language": "en",
          "@value": "Configuration Management Database (CMDB)",
        },
      ],
      "http://www.w3.org/ns/adms#status": Array [
        Object {
          "@id": "https://example.org/def/asset-status/inProduction",
        },
      ],
      "http://www.w3.org/ns/dcat#contactPoint": Array [
        Object {
          "@id": "https://example.org/contact-point/document-management-unit",
        },
      ],
      "http://www.w3.org/ns/dcat#qualifiedRelation": Array [
        Object {
          "@id": "https://example.org/relation/cmdb-gets-data-from-codict",
        },
      ],
      "http://www.w3.org/ns/prov#qualifiedAttribution": Array [
        Object {
          "@id": "https://example.org/attribution/cmdb-application-owner",
        },
      ],
      "https://example.org/def/asset-relation/getsDataFrom": Array [
        Object {
          "@id": "https://example.org/app/codict",
        },
      ],
    },
    "https://example.org/attribution/cmdb-data-data-owner": Object {
      "@id": "https://example.org/attribution/cmdb-data-data-owner",
      "@type": Array [
        "http://www.w3.org/ns/prov#Attribution",
      ],
      "http://www.w3.org/ns/dcat#hadRole": Array [
        Object {
          "@id": "https://example.org/def/agent-role/dataOwner",
        },
      ],
      "http://www.w3.org/ns/prov#agent": Array [
        Object {
          "@id": "https://example.org/org/itec-support",
        },
      ],
    },
    "https://example.org/contact-point/servicenow-support": Object {
      "@id": "https://example.org/contact-point/servicenow-support",
      "@type": Array [
        "http://www.w3.org/2006/vcard/ns#Kind",
      ],
      "http://www.w3.org/2006/vcard/ns#fn": Array [
        Object {
          "@language": "en",
          "@value": "ServiceNow Support",
        },
      ],
      "http://www.w3.org/2006/vcard/ns#hasEmail": Array [
        Object {
          "@id": "mailto:servicenow-support@example.org",
        },
      ],
    },
    "https://example.org/dataset/cmdb-data": Object {
      "@id": "https://example.org/dataset/cmdb-data",
      "@type": Array [
        "http://www.w3.org/ns/dcat#Dataset",
      ],
      "http://purl.org/dc/terms/accrualPeriodicity": Array [
        Object {
          "@id": "https://example.org/def/frequency/continuous",
        },
      ],
      "http://purl.org/dc/terms/created": Array [
        Object {
          "@type": "http://www.w3.org/2001/XMLSchema#date",
          "@value": "2023-08-01",
        },
      ],
      "http://purl.org/dc/terms/description": Array [
        Object {
          "@language": "en",
          "@value": "CMDB stays for Configuration Management Database.",
        },
      ],
      "http://purl.org/dc/terms/issued": Array [
        Object {
          "@type": "http://www.w3.org/2001/XMLSchema#date",
          "@value": "2023-08-01",
        },
      ],
      "http://purl.org/dc/terms/modified": Array [
        Object {
          "@type": "http://www.w3.org/2001/XMLSchema#date",
          "@value": "2024-10-28",
        },
      ],
      "http://purl.org/dc/terms/title": Array [
        Object {
          "@language": "en",
          "@value": "Data from CMDB (Ticketing tool handling requests for IT equipment)",
        },
      ],
      "http://purl.org/dc/terms/type": Array [
        Object {
          "@id": "https://example.org/def/dataset-type/master-data",
        },
      ],
      "http://www.w3.org/ns/adms#status": Array [
        Object {
          "@id": "https://example.org/def/asset-status/inProduction",
        },
      ],
      "http://www.w3.org/ns/dcat#contactPoint": Array [
        Object {
          "@id": "https://example.org/contact-point/servicenow-support",
        },
      ],
      "http://www.w3.org/ns/dcat#qualifiedRelation": Array [
        Object {
          "@id": "https://example.org/relation/cmdb-data-stored-by-cmdb",
        },
      ],
      "http://www.w3.org/ns/dcat#theme": Array [
        Object {
          "@id": "https://example.org/def/business-area/informationTechnology",
        },
      ],
      "http://www.w3.org/ns/prov#qualifiedAttribution": Array [
        Object {
          "@id": "https://example.org/attribution/cmdb-data-data-owner",
        },
      ],
      "https://example.org/def/asset-relation/storedBy": Array [
        Object {
          "@id": "https://example.org/app/cmdb",
        },
      ],
      "https://example.org/def/epdp#confidentialityLevel": Array [
        Object {
          "@id": "https://example.org/def/confidentiality-level/snc",
        },
      ],
    },
    "https://example.org/def/agent-role/dataOwner": Object {
      "@id": "https://example.org/def/agent-role/dataOwner",
      "@type": Array [
        "http://www.w3.org/ns/dcat#Role",
        "http://www.w3.org/2004/02/skos/core#Concept",
      ],
      "http://www.w3.org/2000/01/rdf-schema#label": Array [
        Object {
          "@language": "en",
          "@value": "data owner",
        },
      ],
      "http://www.w3.org/2004/02/skos/core#inScheme": Array [
        Object {
          "@id": "https://example.org/def/agent-role",
        },
      ],
      "http://www.w3.org/2004/02/skos/core#preflabel": Array [
        Object {
          "@language": "en",
          "@value": "data owner",
        },
      ],
    },
    "https://example.org/def/asset-status/inProduction": Object {
      "@id": "https://example.org/def/asset-status/inProduction",
      "@type": Array [
        "http://www.w3.org/2004/02/skos/core#Concept",
      ],
      "http://www.w3.org/2004/02/skos/core#inScheme": Array [
        Object {
          "@id": "https://example.org/def/asset-status",
        },
      ],
      "http://www.w3.org/2004/02/skos/core#prefLabel": Array [
        Object {
          "@language": "en",
          "@value": "In production",
        },
      ],
    },
    "https://example.org/def/business-area/informationTechnology": Object {
      "@id": "https://example.org/def/business-area/informationTechnology",
      "@type": Array [
        "http://www.w3.org/2004/02/skos/core#Concept",
      ],
      "http://www.w3.org/2004/02/skos/core#inScheme": Array [
        Object {
          "@id": "https://example.org/def/business-area",
        },
      ],
      "http://www.w3.org/2004/02/skos/core#prefLabel": Array [
        Object {
          "@language": "en",
          "@value": "Information Technology",
        },
      ],
    },
    "https://example.org/def/confidentiality-level/snc": Object {
      "@id": "https://example.org/def/confidentiality-level/snc",
      "@type": Array [
        "http://www.w3.org/2004/02/skos/core#Concept",
      ],
      "http://www.w3.org/2004/02/skos/core#inScheme": Array [
        Object {
          "@id": "https://example.org/def/confidentiality-level",
        },
      ],
      "http://www.w3.org/2004/02/skos/core#prefLabel": Array [
        Object {
          "@language": "en",
          "@value": "Sensitive non-confidential",
        },
      ],
    },
    "https://example.org/def/dataset-type/master-data": Object {
      "@id": "https://example.org/def/dataset-type/master-data",
      "@type": Array [
        "http://www.w3.org/2004/02/skos/core#Concept",
      ],
      "http://www.w3.org/2004/02/skos/core#inScheme": Array [
        Object {
          "@id": "https://example.org/def/dataset-type",
        },
      ],
      "http://www.w3.org/2004/02/skos/core#prefLabel": Array [
        Object {
          "@language": "en",
          "@value": "Master Data",
        },
      ],
    },
    "https://example.org/def/frequency/continuous": Object {
      "@id": "https://example.org/def/frequency/continuous",
      "@type": Array [
        "http://www.w3.org/2004/02/skos/core#Concept",
      ],
      "http://www.w3.org/2004/02/skos/core#inScheme": Array [
        Object {
          "@id": "https://example.org/def/frequency",
        },
      ],
      "http://www.w3.org/2004/02/skos/core#prefLabel": Array [
        Object {
          "@language": "en",
          "@value": "continuous",
        },
      ],
    },
    "https://example.org/org/itec-support": Object {
      "@id": "https://example.org/org/itec-support",
      "@type": Array [
        "http://www.w3.org/ns/org#Organization",
      ],
      "http://www.w3.org/2004/02/skos/core#prefLabel": Array [
        Object {
          "@language": "en",
          "@value": "ITEC - SUPPORT",
        },
      ],
    },
    "https://example.org/relation/cmdb-data-stored-by-cmdb": Object {
      "@id": "https://example.org/relation/cmdb-data-stored-by-cmdb",
      "@type": Array [
        "http://www.w3.org/ns/dcat#Relationship",
      ],
      "http://purl.org/dc/terms/relation": Array [
        Object {
          "@id": "https://example.org/app/cmdb",
        },
      ],
      "http://purl.org/dc/terms/type": Array [
        Object {
          "@id": "https://example.org/def/asset-relation/storedBy",
        },
      ],
    },
  },
}
`;
